commands/checkout.go

```diff
@@ -97,9 +102,26 @@ func transformCheckoutArgs(args *Args) error {
 		args.Before("git", "remote", "add", "-f", "-t", branch, user, u)
 	}
 
-	idx := args.IndexOfParam(checkoutURL)
-	args.RemoveParam(idx)
-	args.InsertParam(idx, "--track", "-B", newBranchName, fmt.Sprintf("%s/%s", user, branch))
+	remoteName := fmt.Sprintf("%s/%s", user, branch)
+	replaceCheckoutParam(args, checkoutURL, newBranchName, remoteName)
 
 	return nil
 }
+
+func sanitizeCheckoutFlags(args *Args) error {
+	if i := args.IndexOfParam("-b"); i != -1 {
+		return fmt.Errorf("Unsupported flag -b when checking out pull request")
```

Comment:
It's a good start. In the future we can decide whether we want to support something like this syntax:

```
hub checkout -b LOCAL-BRANCH PULLREQ-URL
```

as an alternative for what we have currently:

```
hub checkout PULLREQ-URL LOCAL-BRANCH
```

But it's not worth worrying about now, especially when I want to rething how this whole command works for the next release.